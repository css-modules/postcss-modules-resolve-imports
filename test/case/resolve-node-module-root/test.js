'use strict';

const path = require('path');
const setup = require('../../setup');
const resolveImports = require('../../../index');
 
describe('resolve-node-module root', () => {
  test('defaults to dirname(file)', () => {
    const {resulting: resulting1} = setup(
      require('postcss-icss-values'),
      'local-by-default',
      'extract-imports',
      'scope',
      'self'
    )(__dirname, 'source.css');

    const {resulting: resulting2} = setup(
      require('postcss-icss-values'),
      'local-by-default',
      'extract-imports',
      'scope',
      'self'
    )(__dirname, path.join('node_modules', 'button', 'button.css'));

    expect(resulting1).toMatchSnapshot();
    expect(resulting2).toMatchSnapshot();
  });
  
  /*
   * run process() on entry-point css-files in different directories, both of
   * which import the same file 'node_modules/button/background.css'
   *
   * it is important that the locally scoped classnames in the included
   * 'node_modules/button/background.css' remain the same, no matter
   * which entry point is used (source.css or button.css)
   *
   * (always use relative path of file to rootDir, which is cwd)
   */

  test('process.cwd', () => {
    const root = process.cwd();
    
    const {resulting: resulting1} = setup(
      require('postcss-icss-values'),
      'local-by-default',
      'extract-imports',
      'scope',
      resolveImports({
        root: root,
      }),
    )(__dirname, 'source.css');

    const {resulting: resulting2} = setup(
      require('postcss-icss-values'),
      'local-by-default',
      'extract-imports',
      'scope',
      resolveImports({
        root: root,
      }),
    )(__dirname, path.join('node_modules', 'button', 'button.css'));

    expect(resulting1).toMatchSnapshot();
    expect(resulting2).toMatchSnapshot();
  });
});
