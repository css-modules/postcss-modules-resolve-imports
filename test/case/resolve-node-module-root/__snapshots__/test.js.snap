// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolve-node-module root defaults to dirname(file) 1`] = `
"/*  imported from node_modules/button/background.css  */

._background_background {
  background: red;
}

/*  imported from node_modules/button/button.css  */

._button_button {
  font-size: 1em;
}

/*  imported from source.css  */

._source_test {
  color: white;
}
"
`;

exports[`resolve-node-module root defaults to dirname(file) 2`] = `
"/*  imported from background.css  */

._background_background {
  background: red;
}

/*  imported from button.css  */

._button_button {
  font-size: 1em;
}
"
`;

exports[`resolve-node-module root process.cwd 1`] = `
"/*  imported from test/case/resolve-node-module-root/node_modules/button/background.css  */

._background_background {
  background: red;
}

/*  imported from test/case/resolve-node-module-root/node_modules/button/button.css  */

._button_button {
  font-size: 1em;
}

/*  imported from test/case/resolve-node-module-root/source.css  */

._source_test {
  color: white;
}
"
`;

exports[`resolve-node-module root process.cwd 2`] = `
"/*  imported from test/case/resolve-node-module-root/node_modules/button/background.css  */

._background_background {
  background: red;
}

/*  imported from test/case/resolve-node-module-root/node_modules/button/button.css  */

._button_button {
  font-size: 1em;
}
"
`;
